storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -185,19 +186,26 @@ func (d *DotGit) ObjectPack(hash plumbing.Hash) (billy.File, error) {
 	return pack, nil
 }
 
+// ObjectPack returns a fs.File of the given packfile
+func (d *DotGit) ObjectPack(hash plumbing.Hash) (billy.File, error) {
+	return d.objectPackOpen(hash, `pack`)
+}
+
 // ObjectPackIdx returns a fs.File of the index file for a given packfile
 func (d *DotGit) ObjectPackIdx(hash plumbing.Hash) (billy.File, error) {
-	file := d.fs.Join(objectsPath, packPath, fmt.Sprintf("pack-%s.idx", hash.String()))
-	idx, err := d.fs.Open(file)
-	if err != nil {
-		if os.IsNotExist(err) {
-			return nil, ErrPackfileNotFound
-		}
+	return d.objectPackOpen(hash, `idx`)
+}
 
-		return nil, err
+func (d *DotGit) DeleteObjectPackAndIndex(hash plumbing.Hash) error {
+	err := d.fs.Remove(d.objectPackPath(hash, `pack`))
```

Comment:
I think there should be a minimum age requirement for deletion here, just as with GCing individual objects, to make sure we don't delete a pack that someone else has just uploaded as part of a push, which might not have been referenced yet.